### DISCUSSION TITLE
Allow using same method as setget setter for multiple properties

### DISCUSSION THREAD

**Discussion by mrkdji** [godotengine/godot-proposals#6120] [Scripting]
I was working on a tool node with multiple exported variables and I wanted to send a signal (to update some graphics in the editor) whenever a variable's value is changed. 
As of now I am forced to create a method for each variable as _I am forced to update_ a variable value manually once I define a setter and _have no way to tell_ which variable is being updated:

```gdscript
tool
signal value_changed;

export var a = 0 setget set_a;
export var b = 0 setget set_b;
export var c = 0 setget set_c;

func set_a(v):
  a = v;
  emit_signal("value_changed");

func set_b(v):
  b = v;
  emit_signal("value_changed");

func set_c(v):
  c = v;
  emit_signal("value_changed");
```

I tried to use the **_set** method but it's not getting called in tool mode.

One possible solution could be to modify the setter function to allow 2 parameters, where the second one is the name of the script variable (in the same fashion of _set), that would make possible to do something like this:

```gdscript
tool
signal value_changed;

export var a = 0 setget set_val;
export var b = 0 setget set_val;
export var c = 0 setget set_val;

func set_val(v, n):
  self[n] = v;
  emit_signal("value_changed");
```

**Comment by dalexeev** [comment]
In 4.0 this is possible:

```gdscript
var a:
    get: return get_value("a")
    set(value): set_value("a", value)

var b:
    get: return get_value("b")
    set(value): set_value("b", value)

var _values = {}

func get_value(key):
    return _values.get(key)

func set_value(key, value):
    _values[key] = value

```
